pints/_log_priors.py

```diff
@@ -403,47 +423,52 @@ def n_parameters(self):
 
     def sample(self, n=1):
         """ See :meth:`LogPrior.sample()`. """
-        samples = self._t.rvs(size=n)
-        resample = samples <= 0
-        n_resample = np.sum(resample)
-        while n_resample:
-            samples[resample] = self._t.rvs(size=n_resample)
-            resample = samples <= 0
-            n_resample = np.sum(resample)
-        return samples
+        return scipy.stats.halfcauchy.rvs(loc=self._location,
+                                          scale=self._scale, size=n)
 
 
 class MultivariateGaussianLogPrior(pints.LogPrior):
     """
     Defines a multivariate Gaussian (log)prior with a given ``mean`` and
-    ``covariance`` matrix.
+    covariance matrix ``cov``, with pdf
+
+    .. math::
+        f(x|\\text{mean},\\text{cov}) = \\frac{1}{(2\\pi)^{d/2}|
+        \\text{cov}|^{1/2}} \\text{exp}\\left(-\\frac{1}{2}(x-\\text{mean})'
+        \\text{cov}^{-1}(x-\\text{mean})\\right)
+
+    and expectation
+
+    .. math::
+        \\mathrm{E}(X)=\\text{mean}.
 
-    For example::
+    For example, to create a prior with zero mean and identity covariance,
+    use::
 
-        p = MultivariateGaussianLogPrior(
+        p = pints.MultivariateGaussianLogPrior(
                 np.array([0, 0]), np.array([[1, 0],[0, 1]]))
 
     *Extends:* :class:`LogPrior`
     """
-    def __init__(self, mean, covariance):
+    def __init__(self, mean, cov):
         # Check input
         mean = pints.vector(mean)
-        covariance = np.array(covariance, copy=True)
-        covariance.setflags(write=False)
-        if covariance.ndim != 2:
+        cov = np.array(cov, copy=True)
+        cov.setflags(write=False)
```

Comment:
If we're not returning self._cov anywhere the `setflags(write=False)` is not necessary

Reply:
Done